clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py

```diff
@@ -0,0 +1,586 @@
+    df_clinical_complete = df_clinical_complete.merge(
+        df_clinical[
+            [
+                "blinded_code",
+                "blinded_site",
+                "visit",
+                "diagnosis",
+                "ftld-cdr-global",
+                "cdr-sob",
+            ]
+        ],
+        how="inner",
+        on=["blinded_code", "blinded_site", "visit"],
+    )
+    return df_clinical_complete
+
+
+def dataset_to_bids(complete_data_df: DataFrame, gif: bool) -> Dict[str, DataFrame]:
+    """Selects the data needed to write the participants, sessions, and scans tsvs.
+
+    Parameters
+    ----------
+    complete_data_df: DataFrame
+        Dataframe containing the merged data extracted from the raw images and the clinical data
+
+    gif: bool
+        If True, indicates the user wants to have the values of the gif parcellation
+
+    Returns
+    -------
+    Dict[str, DataFrame]
+        Dictionary containing as key participants, sessions and scans, and the values wanted for each tsv
+    """
+    import os
+
+    # generates participants, sessions and scans tsv
+    complete_data_df = complete_data_df.set_index(
+        ["participant_id", "session_id", "modality", "bids_filename"],
+        verify_integrity=True,
+    )
+    # open the reference for building the tsvs:
+    path_to_ref_csv = os.path.join(
+        os.path.dirname(os.path.dirname(os.path.dirname(__file__))),
+        "data",
+        "genfi_ref.csv",
+    )
+    df_ref = pd.read_csv(path_to_ref_csv, sep=";")
+    if not gif:
+        print(df_ref.head(8))
+    return {
+        col: complete_data_df.filter(items=list(df_ref[col]))
+        for col in ["participants", "sessions", "scans"]
+    }
+
+
+def intersect_data(
+    imaging_data: DataFrame, df_clinical_complete: DataFrame
+) -> DataFrame:
+    """This function merges the dataframe containing the data extracted from the raw images and from the clinical data
+
+    Parameters
+    ----------
+    imaging_data: DataFrame
+        Dataframe containing the data extracted from the raw images
+
+    df_clinical_complete: DataFrame
+        Dataframe containing the clinical data
+
+    Returns
+    -------
+    df_complete: DataFrame
+        Dataframe containing the merged data
+    """
+    df_complete = imaging_data.merge(
+        df_clinical_complete,
+        how="inner",
+        left_on=["source_id", "source_ses_id"],
+        right_on=["blinded_code", "visit"],
+    )
+    df_complete = df_complete.loc[:, ~df_complete.columns.duplicated()]
+    return df_complete
+
+
+def read_imaging_data(source_path: PathLike) -> DataFrame:
+    """This function finds the imaging data and filters it.
+
+    Parameters
+    ----------
+    source_path: PathLike
+        Path to the raw data
+
+    Returns
+    -------
+    df_dicom: DataFrame
+        Dataframe containing the data extracted.
+    """
+    df_dicom = pd.DataFrame(find_dicoms(source_path), columns=["source_path", "source"])
```

Comment:
You can get rid of the intermediate variable : 

```python
return filter_dicoms(
    pd.DataFrame(find_dicoms(source_path), columns=["source_path", "source"])
)

Reply:
nice !